config.yaml

```diff
@@ -1,9 +1,15 @@
+# In order to enable environmental substitution, it is necessary to include all potential configurations in this
+# configuration file. Otherwise, the environment variables will not be automatically added to the ResourceConfig
+# (`mapstructure:",remain"`).
+
+loglevel: info
+
 source:
-  type: filesystem
-  path: /tmp/source.txt
+  type:
```

Comment:
I somehow don't like mixing up the config for the different types on the same level as the user might get confused on what is required for the particular type. We should at least make sure to document which properties are required for the different types somewhere. 

Another alternative would be to introduce an additional level, like:
```yaml
source:
  type: filesystem
  filesystem:
    path: /tmp/foo.txt
  vault:
    token:
    address:
```

What do you think?

Reply:
Thats a good idea. Let's do that ;-)